checkov/terraform/checks/resource/aws/ECRPolicy.py

```diff
@@ -27,13 +27,25 @@ def scan_resource_conf(self, conf):
                 statement = policy['Statement'][0]
                 if statement['Principal'] and type(statement['Principal']) is str:
                     principal = statement['Principal']
-                    if principal == "*":
+                    if principal == "*"  and self.check_for_constrained_condition(statement) == False:
                         self.evaluated_keys = ["policy/Statement/Principal"]
                         return CheckResult.FAILED
         return CheckResult.PASSED
 
     def get_evaluated_keys(self) -> List[str]:
         return ['policy']
 
+    def check_for_constrained_condition(self, statement):
```

Comment:
also a test for terraform would be great

Reply:
Have added a test for pass and fail on constrained condition

Reply:
Has been added :)